worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/PartitionDataWriter.java

```diff
@@ -416,7 +416,13 @@ public void write(ByteBuf data) throws IOException {
       }
 
       data.retain();
-      flushBuffer.addComponent(true, data);
+      try {
+        flushBuffer.addComponent(true, data);
+      } catch (OutOfMemoryError oom) {
```

Comment:
May be Throwable will be better?

Reply:
This makes no sense because the flush buffer has some shuffle data that was received earlier, and then OOM happens. The shuffle data is lost even after committing files.
This would cause data correctness problems here. 
That's why we never try to resume Celeborn workers after OOM happens.

Reply:
Why the shuffle data would be lost?

Reply:
After some investigation, there won't be shuffle data lost.